telemetry/tracks/class-tracks-client.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * Telemetry: Tracks Client class
+ *
+ * @package Automattic\VIP\Telemetry\Tracks
+ */
+
+declare(strict_types=1);
+
+namespace Automattic\VIP\Telemetry\Tracks;
+
+use WP_Error;
+use WP_Http;
+use function Automattic\VIP\Logstash\log2logstash;
+
+/**
+ * Handles all operations related to the Tracks API.
+ */
+class Tracks_Client {
+	/**
+	 * Tracks REST API endpoint for post requests
+	 */
+	protected const TRACKS_ENDPOINT = 'https://public-api.wordpress.com/rest/v1.1/tracks/record';
+
+	/**
+	 * @var WP_Http
+	 */
+	private $http;
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct( WP_Http $http = null ) {
+		$this->http = $http ?? _wp_http_get_object();
+	}
+
+	/**
+	 * Record a batch of events using the Tracks REST API
+	 * 
+	 * @param Tracks_Event[] $events Array of Tracks_Event objects to record
+	 * @return bool|WP_Error True if batch recording succeeded.
+	 *                       WP_Error is any error occured.
+	 */
+	public function batch_record_events( array $events, array $common_props = [] ) {
+		// filter out invalid events
+		$valid_events = array_filter( $events, function ( $event ) {
+			return $event instanceof Tracks_Event && $event->is_recordable() === true;
+		} );
+
+		// no events - nothing to do
+		if ( [] === $valid_events ) {
+			return true;
+		}
+
+		// @codeCoverageIgnoreStart
+		if ( _wp_http_get_object() === $this->http && 'wptests_capabilities' === wp_get_current_user()->cap_key ) {
+			throw new \Exception( 'WP_Http should be mocked in unit tests' );
+		}
+		// @codeCoverageIgnoreEnd
```

Comment:
I'm down to drop this whole part, unless there are objections.

Reply:
Good with me